helix-core/src/comment.rs

```diff
@@ -9,6 +9,29 @@ use crate::{
 use helix_stdx::rope::RopeSliceExt;
 use std::borrow::Cow;
 
+pub const DEFAULT_COMMENT_TOKEN: &str = "//";
+
+/// Returns the longest matching comment token of the given line (if it exists).
+pub fn get_comment_token<'a>(
+    text: RopeSlice,
+    tokens: &'a [String],
+    line_num: usize,
+) -> Option<&'a str> {
+    let mut used_token: Option<&'a str> = None;
+
+    let line = text.line(line_num);
+    let start = line.char_to_byte(line.first_non_whitespace_char()?);
+
+    for token in tokens {
+        let end = std::cmp::min(start + token.len(), line.len_bytes());
+        if line.byte_slice(start..end) == *token {
```

Comment:
```suggestion
        if line.byte_slice(start..).starts_with(token) {
```

one more thing I just noticed: slicing with this `end` might panic if the number of bytes of the comment token doesn't line up with the char boundaries in the text. For example if you have a line that starts with `··` in a Rust file, when you check `///` or `//!` you get a panic because the comment tokens are 3 bytes but ecah `·` is two bytes, so end lies between char boundaries

Reply:
(`starts_with` uses `get_byte_slice` which avoids the panic)

Reply:
Actually with this change we might as well remove the call to `char_to_byte` when binding `start` and use `slice(start..)` instead

Reply:
Hm... I'm seeing a little problem here: Assuming the line comments of Rust are saved as follows:
```toml
["///", "//"]
```
due to `start._with`, `//` is going to be accepted as well, even if we are inside a `///` doc comment, so `//` would be added first.
Are you fine with also checking the length of the token?

Reply:
Ah good point. Instead let's use `get_byte_slice` and `is_some_and(|s| s == *token)` on the start..end byte slice. That will avoid the panic too but be a proper equality check

Reply:
I implemented another solution: Is that ok for you as well?

Reply:
I think that this is worth for a test (checking the length of the token).